landlab/grid/create.py

```diff
@@ -264,106 +413,77 @@ def create_grid(file_like):
            [  7.06,  15.91,  21.5 ,  25.64,  31.55],
            [ 11.55,  17.91,  24.57,  30.3 ,  35.87]])
     """
-    # part 0, parse input
     if isinstance(file_like, dict):
-        dict_like = file_like
+        params = file_like
     else:
-        dict_like = load_params(file_like)
+        params = load_params(file_like)
 
-    # part 1 create grid
-    grid_dict = dict_like.pop("grid", None)
-    if grid_dict is None:
-        msg = "create_grid: no grid dictionary provided. This is required."
-        raise ValueError(msg)
+    if section:
+        grids = params[section]
+    else:
+        grids = params
 
-    for grid_type in grid_dict:
-        if grid_type in _MODEL_GRIDS:
-            grid_class = _MODEL_GRIDS[grid_type]
-        else:
-            msg = "create_grid: provided grid type not supported."
-            raise ValueError
+    new_grids = []
+    for grid_type, grid_desc in as_list_of_tuples(grids):
+        grid_desc = norm_grid_description(grid_desc)
```

Comment:
Should there be a way that a user can indicate at this point which grids get which grid IDs?

Related: 
Also, since this supports having multiple, grids, where are the getting stored. 

`self._grids`??? with a critical plural.

I guess there also needs to be a way for the input file to indicate which grid goes with which component.

Reply:
The newly created grids are returned in a list in the order they are read from the *yaml* file. Grids don't have an ID apart from their index in the list, which depends on how they are listed in the *yaml* file.

Ultimately, I'm not sure how useful being able to describe multiple grids in a single file really is. If one wants to describe multiple components, each with their own grid, they probably should be broken up into separate files where each file has just one grid.

Reply:
Does that imply that create_grid should only be able to create one grid?

Reply:
That's what I'm wondering. For now, I think it's good to have something that can read multiple grids but perhaps we should think about if we really need that functionality or not.

Reply:
Perhaps a better way to think about is that `create_grid` is something that can create multiple grids from a single file, but that the `bmi_bridge` checks that only one grid is provided?

I think the general functionality of `create_grid` makes sense.